pupil_src/shared_modules/video_capture/file_backend.py

```diff
@@ -360,6 +374,18 @@ def toggle_looping(val):
         self.menu.append(ui.Text_Input('frame_num', label='Number of frames',
                                        setter=lambda x: None,
                                        getter=lambda: self.get_frame_count()))
+
+        def toggle_buffering(val):
+            if val and not self.buffering:
+                self.buffered_decoder = self.container.get_buffered_decoder(self.video_stream, dec_batch=50,
+                                                                            dec_buffer_size=200)
+                self.next_frame = self.buffered_decoder.get_frame()
+                self.buffering = True
+            elif not val and self.buffering:
+                self.next_frame = self._next_frame()
+                self.buffering = False
+
+        self.menu.append(ui.Switch('buffering', self, label='Decoder Buffering', setter=toggle_buffering))
```

Comment:
please remove this  switch from UI. Instead can we buffer always except when the condec is mjpeg ?